[rllib] Use compress observations where replay buffers and image obs are used  in tuned examples

avnishn opened the pull request:
Many of our tuned examples using replay and atari wouldn't be able to run on clusters or single node machines because they'll oom without enabling this compression. 

Signed-off-by: avnish <avnish@anyscale.com>

<!-- Thank you for your contribution! Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

gjoliver reviewed (changes requested):
nit nit nit
can we write the comments more formally, like:
```Compress observations to make sure the size of the ReplayBuffer fits most computers. ```
something like that? not really good at wording, but you get the idea.
thanks.

avnishn commented:
could I change 
"we should set this because few machines would be able to contain the replay buffers in memory otherwise" to 

"we should **set compress_observations to True** because few machines would be able to contain the replay buffers in memory otherwise"

is that descriptive enough?

avnishn commented:
I addressed the comments, but during today's experiments realized that only pong examples need obs compression so I turned off the compression in the non pong examples that I changed before.

avnishn commented on rllib/tuned_examples/apex_dqn/atari-apex-dqn.yaml:
```diff
@@ -27,4 +27,4 @@ apex-breakoutnoframeskip-v4:
         target_network_update_freq: 50000
-        min_sample_timesteps_per_iteration: 25000
\ No newline at end of file
+        min_sample_timesteps_per_iteration: 25000
```
just needed a newline

gjoliver reviewed (approved):
this is good trick.

gjoliver commented:
@richardliaw please merge
